### PR TITLE
[Spark-14138][SQL][master] Fix generated SpecificColumnarIterator code can exceed JVM size limit for cached DataFrames

### PR DESCRIPTION
**Pull request by kiszk**
## What changes were proposed in this pull request?

This PR reduces Java byte code size of method in `SpecificColumnarIterator` by using a approach to make a group for  lot of `ColumnAccessor` instantiations or method calls (more than 200) into a method
## How was this patch tested?

Added a new unit test, which includes large instantiations and method calls, to `InMemoryColumnarQuerySuite`

### CODE REVIEW

**Comment by kiszk** [comment]
Jenkins, retest this please

**Comment by kiszk** [comment]
@davies , created PR for master

**Comment by davies** [comment]
LGTM, merging into master.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/GenerateColumnAccessor.scala [hunk 1]
```diff
@@ -114,6 +114,42 @@ object GenerateColumnAccessor extends CodeGenerator[Seq[DataType], ColumnarItera
       (createCode, extract + patch)
     }.unzip
 
+    /*
+     * 200 = 6000 bytes / 30 (up to 30 bytes per one call))
+     * the maximum byte code size to be compiled for HotSpot is 8000.
```
Comment (tedyu) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/GenerateColumnAccessor.scala]: Can you give me link to the above limit of 8000 ?

The answer to http://stackoverflow.com/questions/11345034/how-many-bytes-of-bytecode-has-a-particular-method-in-java seems to show a different limit.

**Pull request closed without merging**